### PR TITLE
normalize paths with spaces - Fixes #252

### LINKED ISSUE #252: This is a bug in Quinoa -- it stops at the space in username
I was running an apllication which uses Quinoa but i am not able to run while i follow all rules during this run

Screenshot (42)

# Second img

Screenshot (43)

### PR DESCRIPTION
**Pull request by johnaohara**
Fixes #252

### CODE REVIEW

**Review by ia3andy — APPROVED** [review]
Good catch thanks @johnaohara :)

File: deployment/src/main/java/io/quarkiverse/quinoa/deployment/packagemanager/PackageManagerInstall.java [hunk 1]
```diff
@@ -70,14 +70,16 @@ private static Installation resolveInstalledNpmBinary(Path installDirectory) {
...
+        final String platformNPMPath = normalizePath(npmPath.toString());
         final String packageManagerBinary = NODE_BINARY + " " + platformNPMPath;
         return new Installation(platformNodeBinPath, packageManagerBinary);
     }
 
-    public static String convertToWindowsPathIfNeeded(String path) {
-        return PackageManager.isWindows() ? path.replaceAll("/", "\\\\") : path;
+    public static String normalizePath(String path) {
+        path = PackageManager.isWindows() ? path.replaceAll("/", "\\\\") : path;
+        path = path.contains(" ") ? "\"".concat(path).concat("\"") : path;
```
Comment (ia3andy) [inline deployment/src/main/java/io/quarkiverse/quinoa/deployment/packagemanager/PackageManagerInstall.java:81]: @johnaohara shouldn't we do this only on Windows?

On other systems we might want to escape them with `\` instead?

File: deployment/src/test/java/io/quarkiverse/quinoa/test/QuinoaPackageManagerInstallPrependBinaryTest.java [hunk 2]
```diff
@@ -41,7 +41,7 @@ public void testQuinoa() {
     }
 
     private static String convertToWindowsPathIfNeeded(String path) {
-        return isWindows() ? PackageManagerInstall.convertToWindowsPathIfNeeded(path) : path;
+        return isWindows() ? PackageManagerInstall.normalizePath(path) : path;
```
Comment (ia3andy) [inline deployment/src/test/java/io/quarkiverse/quinoa/test/QuinoaPackageManagerInstallPrependBinaryTest.java:44]: This is seems to brake the tests

**Review by ia3andy — CHANGES_REQUESTED** [review]
It seems it requires some changes after all

**Comment by johnaohara** [comment]
I'll take a look

File: deployment/src/main/java/io/quarkiverse/quinoa/deployment/packagemanager/PackageManagerInstall.java (hunk 1 above)
Comment (johnaohara) [reply deployment/src/main/java/io/quarkiverse/quinoa/deployment/packagemanager/PackageManagerInstall.java:81]: escaping the entire path with `"` should work in any env, I tested in bash/sh (linux)  and zsh (mac)

File: deployment/src/test/java/io/quarkiverse/quinoa/test/QuinoaPackageManagerInstallPrependBinaryTest.java (hunk 2 above)
Comment (johnaohara) [reply deployment/src/test/java/io/quarkiverse/quinoa/test/QuinoaPackageManagerInstallPrependBinaryTest.java:43]: The entire command was being normalized, not just the path. I have wrapped just the path to the command in the utility method

**Comment by ia3andy** [comment]
@johnaohara test are failing 🤪

**Comment by johnaohara** [comment]
@ia3andy doh, "Tests passed on my machine":tm: I actually ran them before pushing the last commit :smiley:   I'll take another look, sry for the noise

**Comment by johnaohara** [comment]
@ia3andy can we enable manual triggers for the github pipeline? the integration tests are failing in GH, but are passing on my local dev env.  It would speed up my test cycle if I could run the tests in GH manually on my branch, rather that on the PR

**Comment by ia3andy** [comment]
replaced by #262

**Pull request closed without merging**